# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build fontconfig-build glib-build libpng-build pixman-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal; autoconf; autoheader -f; automake -a -c
./configure --disable-interpreter --disable-static --prefix=/wl

# build
make

# install
install -d $DESTDIR/wl/bin
install util/cairo-trace/cairo-trace $DESTDIR/wl/bin
install -d $DESTDIR/wl/include/cairo
install -m 644 cairo-version.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-deprecated.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-features.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-ft.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-pdf.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-ps.h $DESTDIR/wl/include/cairo
install -m 644 src/cairo-svg.h $DESTDIR/wl/include/cairo
install -m 644 util/cairo-gobject/cairo-gobject.h $DESTDIR/wl/include/cairo
install -d $DESTDIR/wl/lib
install src/.libs/libcairo.so.2.* $DESTDIR/wl/lib/libcairo.so.2
ln -sf libcairo.so.2 $DESTDIR/wl/lib/libcairo.so
install util/cairo-gobject/.libs/libcairo-gobject.so.2.* $DESTDIR/wl/lib/libcairo-gobject.so.2
ln -sf libcairo-gobject.so.2 $DESTDIR/wl/lib/libcairo-gobject.so
install -d $DESTDIR/wl/lib/cairo
install util/cairo-trace/.libs/libcairo-trace.so.0.0.0 $DESTDIR/wl/lib/cairo/libcairo-trace.so
install -d $DESTDIR/wl/lib/pkgconfig
install -m 644 src/cairo.pc $DESTDIR/wl/lib/pkgconfig
install -m 644 src/cairo-ft.pc $DESTDIR/wl/lib/pkgconfig
install -m 644 src/cairo-gobject.pc $DESTDIR/wl/lib/pkgconfig
install -m 644 src/cairo-png.pc $DESTDIR/wl/lib/pkgconfig

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/cairo-build.pc << EOF
Name: cairo
Version: 1.14.12
Description: Cairo drawing library
Requires: $REQUIRES

devel=\\
/wl/include/cairo.h \\
/wl/include/cairo-deprecated.h \\
/wl/include/cairo-directfb.h \\
/wl/include/cairo-features.h \\
/wl/include/cairo-ft.h \\
/wl/include/cairo-gobject.h \\
/wl/include/cairo-pdf.h \\
/wl/include/cairo-ps.h \\
/wl/include/cairo-svg.h \\
/wl/include/cairo-version.h \\
/wl/lib/libcairo.so \\
/wl/lib/libcairo-gobject.so \\
/wl/lib/pkgconfig/cairo.pc \\
/wl/lib/pkgconfig/cairo-ft.pc \\
/wl/lib/pkgconfig/cairo-gobject.pc \\
/wl/lib/pkgconfig/cairo-png.pc

exec=\\
/wl/bin/cairo-trace \\
/wl/lib/libcairo.so.2  \\
/wl/lib/libcairo-gobject.so.2 \\
/wl/lib/cairo/libcairo-trace.so
EOF

exit
--- cairo-1.14.12.orig/build/Makefile.am.changelog
+++ cairo-1.14.12/build/Makefile.am.changelog
@@ -5,6 +5,8 @@
 # segment that we put in 'ChangeLog'.  The old ones are named
 # ChangeLog.pre-X.Y where X.Y is the version number of the major release.
 
+AUTOMAKE_OPTIONS = -Wno-portability
+
 CURR_CHANGELOG_VERSION=$(CAIRO_VERSION_MAJOR).$$(echo "($(CAIRO_VERSION_MINOR)+1)/2*2" | bc)
 # examines $version
 PREV_CHANGELOG_VERSION=$$(if test "x$$(echo "($$version-0.1)*2/2"|bc)" = "x$$(echo "$$version*2/2"|bc)"; \
--- cairo-1.14.12.orig/build/configure.ac.system
+++ cairo-1.14.12/build/configure.ac.system
@@ -111,10 +111,10 @@
 AC_CHECK_FUNCS([ctime_r drand48 flockfile funlockfile getline link strndup])
 
 dnl Check if the runtime platform is a native Win32 host.
-AC_COMPILE_IFELSE([[
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
 #ifdef _WIN32
  choke me
-#endif]], [have_windows=no], [have_windows=yes])
+#endif]])], [have_windows=no], [have_windows=yes])
 
 dnl Possible headers for mkdir
 AC_CHECK_HEADERS([sys/stat.h io.h])
--- cairo-1.14.12.orig/configure.ac
+++ cairo-1.14.12/configure.ac
@@ -13,7 +13,7 @@
 
 AC_CHECK_HEADERS([unistd.h sys/ioctl.h])
 
-AM_INIT_AUTOMAKE([1.11 foreign -Wall no-define no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([1.11 foreign -Wall no-define no-dist-gzip dist-xz subdir-objects])
 AM_SILENT_RULES([yes])
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR]) dnl Workaround for Automake 1.12
 
--- cairo-1.14.12.orig/test/Makefile.am
+++ cairo-1.14.12/test/Makefile.am
@@ -2,6 +2,8 @@
 
 include $(top_srcdir)/test/Makefile.sources
 
+AUTOMAKE_OPTIONS = -Wno-override
+
 SUBDIRS=pdiff .
 
 # Then we have a collection of tests that are only run if certain
